storage/remote/client.go

```diff
@@ -198,11 +212,44 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		err = errors.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
 	if httpResp.StatusCode/100 == 5 {
-		return RecoverableError{err}
+		return RecoverableError{err, defaultBackoff}
+	}
+	if httpResp.StatusCode == 429 {
+		if retryAfterHeader := httpResp.Header.Get("Retry-After"); retryAfterHeader != "" {
+			duration, errRetryAfter := retryAfterDuration(retryAfterHeader)
+			if errRetryAfter != nil {
+				return RecoverableError{errors.Wrap(err, errRetryAfter.Error()), duration}
+			}
+		}
+		return RecoverableError{err, defaultBackoff}
 	}
 	return err
 }
 
+func retryAfterDuration(t string) (durationSecs model.Duration, err error) {
```

Comment:
I don't think we need to return an error here. All of the error handling behavior ends up the same as if the defaultBackoff is used. So in the case of an error just return the same as if there is no header, and simplify logic elsewhere.

Reply:
Done